modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthClientCredentialsDancerBuilder.java

```diff
@@ -18,9 +18,12 @@
   /**
    * @param encodeClientCredentialsInBody If @{code true}, the client id and client secret will be sent in the request body.
    *        Otherwise, they will be sent as basic authentication.
+   *
+   * @deprecated since 4.2.0. Use {@link OAuthClientCredentialsDancerBuilder#withClientCredentialsIn(ClientCredentialsLocation)} instead.
    * 
    * @return this builder
    */
+  @Deprecated
   OAuthClientCredentialsDancerBuilder encodeClientCredentialsInBody(boolean encodeClientCredentialsInBody);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Do not forget to remove this deprecated code someday. ([squid:S1133](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1133))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7676&issues=AWjnfbhIotIXnmsUpaHR&open=AWjnfbhIotIXnmsUpaHR)

[ISSUE_KEY]: AWjnfbhIotIXnmsUpaHR